### ISSUE TITLE
Wrong call to mean() in https://www.tensorflow.org/text/tutorials/text_generation

### ISSUE THREAD

**Issue by louridas** [tensorflow/text#810]
In https://www.tensorflow.org/text/tutorials/text_generation the mean loss is calculated as:

```
mean_loss = example_batch_loss.numpy().mean()
```

However, `example_batch_loss.numpy()` has a single value, as `SparseCategoricalCrossEntropy` applies the `AUTO` reduction strategy:

```
print(example_batch_loss.numpy()) # outputs single scalar value
```

The `AUTO` reduction strategy is in almost all cases (as per https://www.tensorflow.org/api_docs/python/tf/keras/losses/Reduction) `SUM_OVER_BATCH_SIZE`: Scalar SUM divided by number of elements in losses.

That means that `example_batch_loss.numpy()` is already the mean so that the call to `mean()` is redundant.

**Comment by broken** [comment]
You're right. Do you want to create a PR? Otherwise, I can make the update.

**Comment by louridas** [comment]
No need for a PR, more straightforward if you could make the update.
